pkg/client/registration.go

```diff
@@ -211,11 +211,60 @@ func monitorSignerPoolStatus(
 		application.String(),
 	)
 
-	newBlockChan := ethereumChain.BlockCounter().WatchBlocks(ctx)
+	updateTrigger := make(chan interface{})
```

Comment:
What do you think about extracting all subscription logic below to a separate function for better readability:
```
func subscribeStatusUpdateEvents() (chan interface{}, func()) {
    updateTrigger := make(chan interface{})

    // ... create all subscriptions

    unsubscribeAll := func() {
        subscription1.Unsubscribe()
        subscription2.Unsubscribe()

        // ... unsubscribe from the rest
     }

    return updateTrigger, unsubscribeAll
}
```
and use it here:
```
updateTrigger, unsubscribeAll := subscribeStatusUpdateEvents()
defer unsubscribeAll()
```

Reply:
Thanks man! I was thinking about that but couldn't find an elegant way to handle unsubscribe calls.